### PR TITLE
changelog: 3.0.0

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Final piece, hey! :sparkles:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+these components have gained super powers through the adition of `onload` and
+`onunload` hooks. Components can now react to being mounted or not, which makes
+them ideal to implement standalone widgets. This behavior uses [html5
+MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)
+under the hood, so it will work anywhere with a DOM!  Again, this was all
+[Shama](http://twitter.com/shamakry)'s hard work.
+
+#### Test coverage
+`choo` has gained a beaut blanket of tests, courtesy of
+[Todd](https://twitter.com/https://twitter.com/whale_eat_squid) and
```
Comment (bendrucker) [inline CHANGELOG.md]: typo: the url

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,85 @@
...
+[Ben](https://twitter.com/bendrucker/).  We've got server, browser and
+pretty-much-all-browsers-known-to-mankind style testing which should give us a
+pretty good idea if stuff breaks. Neat!
+
+#### Core dump
+Internally we've moved the core of `choo` into a separate package -
+[wayfarer](https://github.com/yoshuawuyts/wayfarer). `choo` is now mere glue
+code around `wayfarer`, `yo-yo` and `sheet-router`. This is good news for folks
+who like `choo`, but don't agree with all decision. Go forth and build your
```
Comment (bendrucker) [inline CHANGELOG.md]: typo: decision -> decisions

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,85 @@
...
+
+### Notable changes
+#### Who's the tiniest of them all?
+`choo` is now `5kb` optimized! That's `2kb` less compared to v2. _Woah, how?_
+We now support [yo-yoify](https://github.com/shama/yo-yoify) which optimizes
+those lil template tags to `document.createElement()` calls. So not only is it
+smaller, creating elements now has no overhead. Pretty nifty eh? Mad shoutout
+to [Shama](http://twitter.com/shamakry) for building this!
+
+#### Cpt. Hook(s)
```
Comment (bendrucker) [inline CHANGELOG.md]: Might be worth Cpt. -> Captain for clarity

**Comment by yoshuawuyts** [comment]
@bendrucker thanks heaps for spotting the errs :sparkles:

**Comment by yoshuawuyts** [comment]
Updated with feedback from https://github.com/yoshuawuyts/choo/issues/124, merging!

**Pull request merged**